tests/job.go

```diff
@@ -111,9 +111,19 @@ func NewJob(name string, cmd, args []string, retry, ttlAfterFinished int32, time
 // which tries to contact the host on the provided port.
 // It expects to receive "Hello World!" to succeed.
 func NewHelloWorldJob(host string, port string) *batchv1.Job {
-	check := []string{fmt.Sprintf(`set -x; ping -c 1 %s; x="$(head -n 1 < <(nc %s %s -i 3 -w 3))"; echo "$x" ; if [ "$x" = "Hello World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`, host, host, port)}
-	job := NewJob("netcat", []string{"/bin/bash", "-c"}, check, JobRetry, JobTTL, JobTimeout)
-	return job
+	check := fmt.Sprintf(`set -x; x="$(head -n 1 < <(nc %s %s -i 3 -w 3))"; echo "$x" ; if [ "$x" = "Hello World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`, host, port)
+	return newHelloWorldJob(check)
+}
+
+// NewHelloWorldJobv6 is a workaround to generate `batch.V1` jobs that will access services exposed via IPv6 addresses.
+// It does exactly what NewHelloWorldJob does, but for IPv6.
+func NewHelloWorldJobv6(host string, port string) *batchv1.Job {
```

Comment:
I would also avoid the duplication between the ipv4 and ipv6 versions.

For example -
```
func newHelloWorldJob(host string, port string, ping boolean) *batchv1.Job {
        var pingCmd string
        if ping {
            pingCmd = fmt.Sprintf("ping -c 1 %s;", host)
        }
}
	check := []string{fmt.Sprintf(`set -x; %s x="$(head -n 1 < <(nc %s %s -i 3 -w 3))"; echo "$x" ; if [ "$x" = "Hello World!" ]; then echo "succeeded"; exit 0; else echo "failed"; exit 1; fi`, pingCmd, host, port)}
	job := NewJob("netcat", []string{"/bin/bash", "-c"}, check, JobRetry, JobTTL, JobTimeout)
	return job
}

func NewHelloWorldJob(host string, port string) *batchv1.Job {
	newHelloWorldJob(host, port, false)
}

func NewHelloWorldJob(host string, port string) *batchv1.Job {
	newHelloWorldJob(host, port, true)
}

```

Reply:
I prefer to avoid using the boolean parameter, and duplicate the functions instead - since a boolean argument implicitly makes a function do one of 2 things, when it is supposed to do a single one.

Hope this is not a deal breaker .

Reply:
You can pass a string with a `preCmd` instead. I really don't like the duplication.
But I won't insist.